extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -94,7 +104,7 @@ void annotationTransformers(BuildProducer<AnnotationsTransformerBuildItem> trans
                     return;
                 }
                 for (DotName annotationName : ctx.getTarget().asClass().annotations().keySet()) {
-                    if (metricsAnnotations.contains(annotationName)) {
+                    if (DotName.createSimple(Gauge.class.getName()).equals(annotationName)) {
```

Comment:
It's really not a good idea to create a new `DotName` for each element.

Same for the line below.

Reply:
Good point, will add one constant for each